src/toniarts/openkeeper/game/controller/GameWorldController.java

```diff
@@ -75,6 +75,7 @@
 import toniarts.openkeeper.tools.convert.map.Tile;
 import toniarts.openkeeper.tools.convert.map.Variable;
 import toniarts.openkeeper.utils.WorldUtils;
+import toniarts.openkeeper.view.selection.SelectionArea;

```

Comment:
We can't mix these like this. High hygiene levels must be maintained, especially during Covid season.

The logic needs to be separate from the view. If some common datatransfer etc. class is needed, it needs to be in another package. The VIEW basically uses logic classes yes, but definitely NOT the other way around.

Reply:
Perhaps the initial location of the SelectionArea class is not very good. It does not contain any visual component. All visual part contains in `toniarts.openkeeper.view.selection.SelectionHandler`

Reply:
I have tossed these common classes to toniarts.openkeeper.common, but also toniarts.openkeeper.game.data might be a place for this. Dunno, but not view if this is going to be used in logic.

And if this goes through network, as slim as possible. All you can generate or calculate, can be done so. Cached with transient etc fields.

Reply:
SelectionArea contains 2 Vector2f fields. Transmission surrendered a bit more expensive

Reply:
Well, if it is the same two fields that already went there, that is ok. This is from user interaction (meaning not all the time). So the extra layer is just few bytes.

Reply:
I generally think we need to use the Point class as often as possible. In most cases when working with a map, we work with tiles array indices, not vectors. SelectionArea also manipulate with indexes (not vectors)

Reply:
I agree using point instead of vectors. Vectors give me the chills. Also LOGIC shouldn't contain jME classes at all (or as little as possible at least). Or was this vector from the Math class or the libGdx... anyway.